xcdat/temporal.py

```diff
@@ -1285,19 +1252,14 @@ def _get_weights(self, time_bounds: xr.DataArray) -> xr.DataArray:
         # or time unit (with rare exceptions see release notes). To avoid this
         # warning please use the scalar types `np.float64`, or string notation.`
         if isinstance(time_lengths.data, Array):
-            time_lengths.load()
+            time_lengths = time_lengths.astype("timedelta64[ns]")
 
         time_lengths = time_lengths.astype(np.float64)
 
         grouped_time_lengths = self._group_data(time_lengths)
         weights: xr.DataArray = grouped_time_lengths / grouped_time_lengths.sum()
         weights.name = f"{self.dim}_wts"
 
-        # Validate the sum of weights for each group is 1.0.
```

Comment:
It seems to be a good feature to have to check if the sum matches. But if it de-gradates the performance a lot, maybe this check can be just implemented in testing (if it is not included yet). Also the `_get_weights` description needs to be updated to reflect that sum is no longer validated.

Reply:
We should expect the logic of `_get_weights()` to be correct, so this assertion should not be necessary at runtime (especially with the performance hit).

I like your suggestion of making it a unit test instead. I will push a commit with this change soon.